src/distributions/other.rs

```diff
@@ -145,6 +147,37 @@ impl Distribution<bool> for Standard {
     }
 }
 
+/// Requires nightly Rust and the [`simd_support`] feature
+///
+/// Note that on some hardware like x86/64 mask operations like [`_mm_blendv_epi8`]
+/// only care about a single bit. This means that you could use uniform random bits
+/// directly:
+///
+/// ```ignore
+/// // this may be faster...
+/// let x = unsafe { _mm_blendv_epi8(a.into(), b.into(), rng.gen::<__m128i>()) };
+///
+/// // ...than this
+/// let x = rng.gen::<mask8x16>().select(b, a);
+/// ```
+///
+/// Since most bits are unused you could also generate only as many bits as you need.
+///
+/// [`_mm_blendv_epi8`]: https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_blendv_epi8&ig_expand=514/
+/// [`simd_support`]: https://github.com/rust-random/rand#crate-features
+#[cfg(feature = "simd_support")]
+impl<T, const LANES: usize> Distribution<Mask<T, LANES>> for Standard
+where
+    T: MaskElement + PartialOrd + SimdElement<Mask = T> + Default,
+    LaneCount<LANES>: SupportedLaneCount,
+    Standard: Distribution<Simd<T, LANES>>,
+{
+    #[inline]
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Mask<T, LANES> {
+        rng.gen().lanes_lt(Simd::default())
+    }
```

Comment:
Doesn't this compute to `false` in all lanes?

And, regarding the proceeding doc, ... doesn't this make the `Mark` type inappropriate if normally only one bit from each lane is used? Ugh, guess we can't fix that.

BTW one can assume that a random boolean is `true` with 50% probability, but should one actually assume anything about what a random `Mask` is?

Maybe instead we should provide a higher level API around `select` etc... though given that SIMD is a little niche, maybe just a few recipes in the book would be enough.

Reply:
No specific use-case; I was just going by what you mentioned (select). Probably this is unnecessarily to get into however.